lib/auth/auth.go

```diff
@@ -289,6 +290,12 @@ func NewServer(cfg *InitConfig, opts ...ServerOption) (*Server, error) {
 			return nil, trace.Wrap(err)
 		}
 	}
+	if cfg.CloudClients == nil {
+		cfg.CloudClients, err = cloud.NewClients()
```

Comment:
I suppose this is unrelated to your change, but I do find it odd that we initialize clients for AWS, GCP, and Azure unconditionally. Feels wasteful.

Reply:
Is your comment on the implementation of `cloud.NewClients()` or the fact that I call it here?

By my reading, calling `cloud.NewClients()` doesn't actually initialize any clients, only a cache of clients that will be initialized when they are used

Reply:
The implementation. But I agree with your assessment that it's just a cache. That makes me feel better.